### PR TITLE
chore:  enhance testing composejs plugin

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
1. Adding an example to test if the plugins is correctly composed.

### CODE REVIEW

File: packages/rolldown/src/rolldown-build.ts [hunk 1]
```diff
@@ -6,11 +6,13 @@ import { createBundler } from './utils/create-bundler'
 import type { RolldownOutput } from './types/rolldown-output'
 import type { HasProperty, TypeAssert } from './utils/type-assert'
 import type { InputOptions } from './options/input-options'
+import { NormalizedInputOptions } from './options/normalized-input-options'
 
 export class RolldownBuild {
   #inputOptions: InputOptions
   #bundler?: Bundler
   #stopWorkers?: () => Promise<void>
+  normalizedInputOptions?: NormalizedInputOptions
```
Comment (hyf0) [inline packages/rolldown/src/rolldown-build.ts]: Not sure about leaking this to outside just for testing.

**Comment by hyf0** [comment]
What if we expose `composeJsPlugins` under `rolldown/expermental` to outside and test it directly?

File: packages/rolldown/src/rolldown-build.ts (hunk 1 above)
Comment (IWANABETHATGUY) [reply packages/rolldown/src/rolldown-build.ts]: that's maybe not the best way to do it, but I don't find other methods to get normalized plugin length.

Comment (hyf0) [reply packages/rolldown/src/rolldown-build.ts]: I believe testing `composeJsPlugins` directly is more proper.

**Comment by IWANABETHATGUY** [comment]
> What if we expose `composeJsPlugins` under `rolldown/expermental` to outside and test it directly?

Sounds like a good idea.

File: scripts/meta/constants.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import * as selfExports from './constants.js'
+// oxlint-disable
```
Comment (IWANABETHATGUY) [inline scripts/meta/constants.js:2]: avoid race condition

**Comment by hyf0** [comment]
We don't need to write everything in fixture test. We still use fixture-based tests to testing if enabling composingPlugins give correct output. But the behavior of `composeJsPlugins` itself could be tested directly in another test files.

**Review by hyf0 — APPROVED** [review]

**Pull request merged**